server/manager/src/main/java/org/apache/accumulo/manager/upgrade/Upgrader11to12.java

```diff
@@ -84,111 +91,102 @@ public void upgradeZookeeper(@NonNull ServerContext context) {
   public void upgradeRoot(@NonNull ServerContext context) {
     log.debug("Upgrade root: upgrading to data version {}", METADATA_FILE_JSON_ENCODING);
     var rootName = Ample.DataLevel.METADATA.metaTable();
-    processReferences(context, rootName);
+    // not using ample to avoid StoredTabletFile because old file ref is incompatible
+    try (AccumuloClient c = Accumulo.newClient().from(context.getProperties()).build();
+        BatchWriter batchWriter = c.createBatchWriter(rootName); Scanner scanner =
```

Comment:
context is already an an instance of AccumuloClient. You don't need to create a new one. I see this is an issue with the existing upgrader code prior to this PR, but it should still be fixed. Funny thing is that the scanner is already created this way.

```suggestion
    try (var batchWriter = context.createBatchWriter(rootName); var scanner =
```

Reply:
Fixed in fe6e9e4172